### DISCUSSION TITLE
Release 0.6.7 (1.21.20.21)

### DISCUSSION THREAD

**Discussion by m-brisch** [Mojang/minecraft-editor#181] [Announcements]
_[Read the previous release notes here](https://github.com/Mojang/minecraft-editor/discussions/180)_  

---    

## Summary  

Hello and happy preview day 🥳! We are now on 0.6.7 (1.21.20.21), as of Thursday, June 27th, 2024. 

As a reminder of Editor API support:   

- Editor Extensions [overview article](https://learn.microsoft.com/en-us/minecraft/creator/documents/editorextensionsintroduction) and [API reference documentation](https://aka.ms/EditorAPI) on learn.microsoft.com  
- [Starter Kit](https://github.com/Mojang/minecraft-editor-extension-starter-kit): Contains the compiler tools and various libraries required to get started making Editor Extensions  
- [Samples](https://github.com/Mojang/minecraft-editor-extension-samples) created by our team to jump-start your own Extension creations  

Additionally, if you're new to the Editor GitHub this week, we encourage you to join the Bedrock Add-Ons Discord [#Editor channel](https://discord.gg/TZnQKFwGpt)!  

--

## New Feature: Deferred Lighting Settings
We added a tool to modify the deferred lighting settings directly within Editor so you can easily fine-tune your world's experience. You can open the tool panel by going to `View => Deferred Lighting Settings` in the top menu. When you're done configuring your world you can hit the `Export to clipboard` button at the bottom of the page which you can then paste directly into your resource pack.

There are two known issues we're currently working to resolve:
  - Known issue: shadows are cast from the UI elements. You can toggle crosshair mode via CTRL+TAB as a workaround
  - Known issue: paste preview (CTRL+SHIFT+V) does not render properly when deferred lighting is enabled. 

For more information about Deferred Lighting, check out the [learn page](https://learn.microsoft.com/en-us/minecraft/creator/documents/deferredlighting?view=minecraft-bedrock-stable).

Have more ideas or features for the Deferred Lighting Settings panel? Create a post here on the `minecraft-editor` GitHub or in the [#Editor channel](https://discord.gg/TZnQKFwGpt)!  

https://github.com/Mojang/minecraft-editor/assets/29804144/3a79dee2-d0c2-498c-b226-3f262436d9ab

--

## New Feature: Custom Input Mapping
We added a tool to allow you to remap the shortcuts used by Editor and all Editor Extensions! From the Action Bar hit `+` and then `Input Mapping` to add the shortcut to your Action Bar. Then hit `Input Mapping` to open the panel.
  - Key bindings can be reconfigured using the shortcut field, which detects key presses to set the binding. Pending changes will highlight the text until saved or cleared. Closing the modal reverts any pending changes.
  - Right-clicking on a selected shortcut field clears the binding, or clear button next to the field can be used.
  - Added a toggle for exclusive (InputModifier.None in API) key actions to support advanced input scenarios.
  - Modified bindings can be restored by clicking restore buttons for the binding, active context, and all modified bindings.
  - Saved bindings are accessible across different editor projects.

![image](https://github.com/Mojang/minecraft-editor/assets/29804144/ea0a348e-ba50-4ffd-97da-921d997aaf75)

---

## Change Log   
- Added settings panel to modify atmospheric, global illumination, and color grading settings for deferred lighting (requires deferred lighting resource pack and settings to be enabled)
  - Known issue: shadows are cast from the UI elements. You can toggle crosshair mode via CTRL+TAB as a workaround
  - Known issue: paste preview (CTRL+SHIFT+V) does not render properly when deferred lighting is enabled. 
- Added Input Mappings action bar item that opens a modal displaying all available key bindings and input contexts. 
  - Key bindings can be reconfigured using the shortcut field, which detects key presses to set the binding. Pending changes will highlight the text until saved or cleared. Closing the modal reverts any pending changes.
  - Right-clicking on a selected shortcut field clears the binding, or clear button next to the field can be used.
  - Added a toggle for exclusive (InputModifier.None in API) key actions to support advanced input scenarios.
  - Modified bindings can be restored by clicking restore buttons for the binding, active context, and all modified bindings.
  - Saved bindings are accessible across different editor projects. 
- Extended game options to include more export options as well as applying options to export level data.
- Added an optional KeyBindingInfo argument to the key binding registration API functions to assign additional information to shortcuts.
- Added new BlockList UI element to Property Pane API.
- Added inputContextId and inputContextLabel optional properties to ModalToolCreationParameters API to represent user defined modal tool identifiers for modal tools.
- Added interface KeyBinding and modified key binding registration API methods to take this object.
- Dropdown menus will attempt to open in the opposite direction when they extend beyond the window bounds.
- Added variant: ColorPickerVariant optional property to IPropertyItemOptionsColorPicker API to support different types of color picker property items. Changed the default variant to respect the pane layout. Moved gradient selector for Default and Inline variants to a popup, accessible by clicking the selected color box.
- Updated set of entities that appear within the Editor picker
- Updated set of blocks that appear within the Editor picker
- Introduced a new filter "is_navigating" that checks to see if the entity is currently pathfinding. This requires a "minecraft:navigation" component

---   

Thank you for being here! You're the best! ❤️

**Comment by AbbottMc** [comment]
Awsome update!
